src/controllers/users-controller.js

```diff
@@ -0,0 +1,127 @@
+const jwt = require("jsonwebtoken");
+require('dotenv').config();
+const SECRET = process.env.SECRET;
+
+const User = require("../models/users-schema");
+
+const { getUser, setUserKey } = require('../services/users-services');
+
+const registerUser = async (req, res, next) => {
+  try {
+    const { email, password } = req.body;
+
+    const user = await getUser({ email: email });
+    if (user) {
+      res
+        .status(409)
+        .json({message: "Email in use" });
+    }
+
+    const newUser = new User({ email });
+    await newUser.setHashPassword(password);
+    await newUser.save();
+    return res
+      .status(201)
+      .json({
+        message: "User registered",
+        user: { email: newUser.email, subscription: newUser.subscription },
+      });
+  } catch (error) {
+    next(error);
+  }
+};
+
+const loginUser = async (req, res, next) => {
+  const { email, password } = req.body;
+  const user = await getUser({ email: email });
+
+  if (!user) {
+    return res
+      .status(401)
+      .json({ message: "No such user" });
+  }
+
+  const isPasswordCorrect = await user.validatePassword(password);
+  if (!isPasswordCorrect) {
+    return res.status(401).json({ message: "Wrong password" });
+  }
+
+  const payload = {
+    id: user.id,
+    email: user.email,
```

Comment:
Potrzebujesz ten email potem? Bo jeśli nie to go nie wysylaj.

Reply:
Ok, niepotrzebny, więc nie wysyłam. Poprawione.